bioimageio/spec/shared/utils.py

```diff
@@ -182,8 +181,10 @@ def transform_ImportableModule(node):
 
     @staticmethod
     def transform_ResolvedImportableSourceFile(node: ResolvedImportableSourceFile) -> nodes.ImportedSource:
+        module_path = resolve_uri(node.source_file)
+        module_name = os.path.splitext(os.path.split(module_path)[1])[0]
         importlib_spec = importlib.util.spec_from_file_location(
-            f"user_imports.{uuid.uuid4().hex}", resolve_uri(node.source_file)
+            module_name, module_path
```

Comment:
this might collide with an existing module...
How is the random module name of  `uuid.uuid4().hex` causing any trouble?

Reply:
if repeated import is an issue, we should maybe use ['uuid.uuid5()`](https://docs.python.org/3/library/uuid.html#uuid.uuid5) ?